app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -107,7 +107,7 @@ public void fetchContributions() {
                     .filter(mwQueryLogEvent -> !mwQueryLogEvent.isDeleted()).doOnNext(mwQueryLogEvent -> Timber.d("Image %s passed filters", mwQueryLogEvent.title()))
                     .map(image -> {
                         Contribution contribution = new Contribution(null, null, image.title(),
-                                "", -1, image.date(), image.date(), user,
+                                "", -1, image.date(), image.date(), user, user,
```

Comment:
this seems wrong in light of the other constructor where you set creator/user to different values
` null, null, sessionManager.getAuthorName(), sessionManager.getUserName(),`

Reply:
This is for fetching current user's contributions and user value is username only

Reply:
Okay so when we are fetching current users contributions 
```
creator == sessionManager.getUserName()
user == sessionManager.getUserName()
```
but when we are creating them to upload
```
creator == sessionManager.getAuthorName()
user == sessionManager.getUserName()
```

That is correct behaviour?

Reply:
Hmm, didn't notice that. we are not displaying author in media detail.  don't know why it is set to username.
@sivaraam shouldn't creator be getAuthorName() in first one too.

Reply:
@sivaraam any insight?

Reply:
Apologies for the slow response.

> Hmm, didn't notice that. we are not displaying author in media detail. don't know why it is set to username.
> @sivaraam shouldn't creator be getAuthorName() in first one too.

Yeah. That sounds right to me (for the sake of consistency). @318anushka seems to have updated it accordingly in 6046c4525328f5a8c05c8afb0fabffd1964cfe40.

Anyways, we don't seem to be showing the Author name in the Contributions activity at all, so I'm not sure how much this matters. Regardless, it's better to be consistent. :)

Reply:
I am not sure if `creator` is a field we ever need right now at this point.

I'll work through this data flow

`sessionManager.getAuthorName()` is the user name we use to login, not an authorName in any way conceivable. It first trys to get a `rawUsername` whatever that is, I am pretty sure that always returns null, then just calls `sessionManager.getUserName()`

When we create a Contribution for upload we set the creator to `sessionManager.getAuthorName()` aka just the user name

a creator is set on a Contribution at the start of uploading when we have a custom author name set 

If the creator field is empty then later on in an upload we set the creator as `sessionManager.getAuthorName()`

We then upload the creator twice??
```
.append("|author=[[User:").append(contribution.getCreator()).append("|")
        .append(contribution.getCreator()).append("]]\n");
```


When we create Contributions from the `logEvents` endpoint we use `sessionManager.getUserName()` which is again still userName

When we turn MwQueryResponses into media
Creator is null when imageInfo or metadata is null
Otherwise our happy little friend `getArtist` aka user


Does this all make sense? Am I taking crazy pills or is creator just supposed to be userName and custom author name should never have cursed this model?

Reply:
From what i understood , creator is useful when we are setting custom author name.
`if (store.getBoolean("useAuthorName", false)) {
            String authorName = store.getString("authorName", "");
            contribution.setCreator(authorName);
        }`

for the rest of the cases it is set to username.

I don't think getArtist always returns user,  it is creator of image which in certain cases might be custom name.
Example:
https://commons.wikimedia.org/w/api.php?action=query&format=json&formatversion=2&prop=imageinfo&iiprop=url|extmetadata|user&iiurlwidth=640&iiextmetadatafilter=DateTime|Categories|GPSLatitude|GPSLongitude|ImageDescription|DateTimeOriginal|Artist|LicenseShortName|LicenseUrl&titles=File%3AFlowe_3.jpg
Here my username is Anushka and getArtist will return asdf.

Also while uploading 
`.append("|author=[[User:").append(contribution.getCreator()).append("|")
        .append(contribution.getCreator()).append("]]\n");`
shouldn't first one of these field be username and then creator. I don't know about this just inferring from [here](https://github.com/commons-app/apps-android-commons/issues/3160#issuecomment-605503067)

Reply:
It probably should be, I am now quite interested in what this artist metadata returns when we actually put the right data into this `pageContents`, this could be a cascade of bugs and that is the actual root.

```
author=[[User:myUserName|myCustomName]]
```
makes me even more suspicious of everything. This `author/creator/artist/user` stew is thoroughly annoying

Reply:
> I am not sure if creator is a field we ever need right now at this point.

To add to what @318anushka said, creator is also useful as some images might literally have no relation between the Author and the person who uploaded it. Here's an example: [File: Troy Hunt](https://commons.wikimedia.org/wiki/File:Troy_Hunt.jpg). The Author is *Troy Hunt* while the user who uploaded it is *IagoQnsi*.

This most likely comes into picture only in Explore, though. It should actually be a concern in Contributions too but let's give ourselves some break and not get into that for now ;)

> When we create a Contribution for upload we set the creator to sessionManager.getAuthorName() aka just the user name

The author name information should come from the media rather than from the session manager but I believe we can't do it right now as we are using the `logevents` API. #2904 would change this, I believe.

> We then upload the creator twice??

Just to clear the confusion, that's just the code for constructing the following wiki text:

```
[[User:USER_NAME | USER_NAME]]
```

The redundant USER_NAME is just to hide the `User:` prefix from the rendered wiki text.

> It probably should be, I am now quite interested in what this artist metadata returns when we actually put the right data into this `pageContents` ...

I really couldn't get this. I think this PR is in a right shape now. If there's something that's still not clear let me know. I'll see if I could help with clarifying that.

> Also while uploading
.append("|author=[[User:").append(contribution.getCreator()).append("|") .append(contribution.getCreator()).append("]]\n");
shouldn't first one of these field be username and then creator. I don't know about this just inferring from here

I have zero knowledge of the rationale behind custom author name. But I'm suspecting it's to enable users to have a custom name in author field (as opposed to allowing them to attribute images created by others) in which case you're right. The first one should be username. Anyways, that should probably be fixed separately.

> This author/creator/artist/user stew is thoroughly annoying

Yeah. This is the mess you get when the data model of the app and that returned by the server don't align well. 🤷‍♂️

Reply:
I am not particularly happy with this PR. Why did we bother extracting the contents of the anchor tag, removing the `page does not exist` suffix only to add and use a new different field?
Why is DeleteHelper still trying to remove `page does not exist`?

This new field gets assigned like `sessionManager.getAuthorName(), sessionManager.getUserName(),` and having these 2 identical damned methods beside each other adds more to confusion making them seem like separate things.

I think adding `user` is a mistake and perhaps the only changes in this PR should be `getArtist` and get rid of the suffix removal in DeleteHelper.

Viewed in browser the author field is, I believe, whatever we sent in `author=[[User:myUserName|userName again?]]`
![image](https://user-images.githubusercontent.com/3358282/79315216-92eab300-7efa-11ea-9dec-4566e9a810e1.png)
And we should not differ from the website

Reply:
Though I guess `uploadedBy` is information that isn't explicitly mentioned anywhere on the website? I'll seek confirmation on the ticket

Reply:
> I am not particularly happy with this PR. Why did we bother extracting the contents of the anchor tag, removing the `page does not exist` suffix only to add and use a new different field?

Because if we don't then the `(page does not exist)` would still show up in the detail view of the images you view via 'Explore'. The new field is to just show the username near the "Uploaded by:". We still show the "Author" in the media detail view.

> Why is DeleteHelper still trying to remove `page does not exist`?
> 

Thanks for bringing that up! We can remove that code now. cc @318anushka.
It really should've been caught during the PR that added that code. Looks like it slipped through.

Also, that's another place where the `user` would be the appropriate thing to use instead of the author/creator. Something to be done in a separate PR :)

> This new field gets assigned like `sessionManager.getAuthorName(), sessionManager.getUserName(),` and having these 2 identical damned methods beside each other adds more to confusion making them seem like separate things.
> 

Yeah. I think we can just go with ` user, user` for that part instead of `sessionManager.getAuthorName(), user`. There's not much difference as of now like you point out.

> I think adding `user` is a mistake and perhaps the only changes in this PR should be `getArtist` and get rid of the suffix removal in DeleteHelper.
>

I would say it's a "boon" for the reasons I mention above. You'll likely get more info about this in the issue thread where you've asked Nicolas for clarification. Let me know if you're still find some mistakes with introducing `user`. I'll see if I could clarify :)

> Viewed in browser the author field is, I believe, whatever we sent in `author=[[User:myUserName|userName again?]]`
> ![image](https://user-images.githubusercontent.com/3358282/79315216-92eab300-7efa-11ea-9dec-4566e9a810e1.png)
> And we should not differ from the website

Do you mean we shouldn't be doing any kind of extraction from anchor tag for the Author? If yes, then I agree. It's best if we do it that way. But it would be a more involved change and is out of scope of this PR.

The HTML fiddling has it's limitations such as the "Author" field not being shown in the detail view for some images. I believe this fiddling with the HTML came into picture as we needed a username that we could use as a simple string for use in other places.

Reply:
one reason it's getting quite confusing is because we are having different names for same field. How about we keep only 2 user and author for now, replacing creator with author. All of this including changes you mentioned above could be done in separate pr after this.
 
Also i was just waiting for nicolas's input, thought to make changes after clarifying.